CRM/ACL/BAO/ACLEntityRole.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+
+/**
+ *  Access Control AclRole.
+ */
+class CRM_ACL_BAO_ACLEntityRole extends CRM_ACL_DAO_ACLEntityRole {
+  public static $_entityTable = NULL;
+
+  /**
+   * Whitelist of possible values for the entity_table field
+   *
+   * @return array
+   */
+  public static function entityTables(): array {
+    if (!self::$_entityTable) {
+      self::$_entityTable = [
+        'civicrm_contact' => ts('Contact'),
+        'civicrm_group' => ts('Group'),
+      ];
+    }
+    return self::$_entityTable;
```

Comment:
Can we take out this silly and pointless cache?
All this function needs to do is return an array - there's no need for a static variable.

Reply:
@colemanw I think this is just a class rename so that tidy up probably should be done separately